ta/arch/arm/user_ta_header.c

```diff
@@ -58,42 +58,19 @@ void ta_entry_invoke_command(uint32_t cmd_id, uint32_t param_types,
 			     TEE_Param params[TEE_NUM_PARAMS],
 			     uint32_t session_id) __noreturn;
 
-/* These externs are defined in the ld link script */
-extern uint32_t linker_RO_sections_size;
-extern uint32_t linker_RW_sections_size;
-extern uint32_t linker_res_funcs_ZI_sections_size;
-extern uint32_t linker_rel_dyn_GOT;
-
-/* Note that cmd_id is not used in a User Mode TA */
-const struct user_ta_func_head user_ta_func_head[]
-			__attribute__ ((section(".ta_func_head"))) = {
-	{ 0, (uint32_t)ta_entry_open_session },
-	{ 0, (uint32_t)ta_entry_close_session },
-	{ 0, (uint32_t)ta_entry_invoke_command },
-	{ (TA_FLAG_USER_MODE | TA_FLAGS), 0 /* Spare */ },
-	{ (TA_DATA_SIZE), (TA_STACK_SIZE) },
-};
-
-const struct user_ta_head ta_head __attribute__ ((section(".ta_head"))) = {
+const struct ta_head ta_head __section(".ta_head") = {
 	/* UUID, unique to each TA */
 	TA_UUID,
-	/* Number of functions in the TA */
-	sizeof(user_ta_func_head) / sizeof(struct user_ta_func_head),
-	/* Section size information */
-	(uint32_t)&linker_RO_sections_size,
-	(uint32_t)&linker_RW_sections_size,
-	(uint32_t)&linker_res_funcs_ZI_sections_size,
-	(uint32_t)&linker_rel_dyn_GOT,
-	/* Hash type, filled in by sign-tool */
-	0,
-	/* TA trace level */
-	/* TA_TRACE_LEVEL_DEFAULT, */
+	TA_STACK_SIZE,
```

Comment:
`TA_DATA_SIZE` has disappeared. Should not it be back as there is a GP property `gpd.ta.dataSize`?